node-graph/gcore/src/text.rs

```diff
@@ -0,0 +1,42 @@
+mod font_cache;
+mod to_path;
+
+use crate::EditorApi;
+pub use font_cache::*;
+pub use to_path::*;
+
+use crate::Node;
+
+pub struct TextGenerator<Text, FontName, Size> {
+	text: Text,
+	font_name: FontName,
+	font_size: Size,
+}
+
+impl<'a: 'input, 'input, Text, FontName, Size> Node<'input, EditorApi<'a>> for TextGenerator<Text, FontName, Size>
+where
+	Text: Node<'input, (), Output = String>,
+	FontName: Node<'input, (), Output = Font>,
+	Size: Node<'input, (), Output = f64>,
+{
+	type Output = crate::vector::VectorData;
+	#[inline]
+	fn eval(&'input self, editor: EditorApi<'a>) -> Self::Output {
+		let text = self.text.eval(());
+		let font = self.font_name.eval(());
+		let font_size = self.font_size.eval(());
+
+		let buzz_face = editor.font_cache.and_then(|cache| cache.get(&font)).map(|data| load_face(data));
+		crate::vector::VectorData::from_subpaths(to_path(&text, buzz_face, font_size, None))
+	}
+}
```

Comment:
Yes, it runs into the error:
```sh
error: implementation of `graphene_core::Node` is not general enough
   --> node-graph/interpreted-executor/src/node_registry.rs:52:5
    |
52  |                 Box::pin(any)
    |                 ^^^^^^^^^^^^^ implementation of `graphene_core::Node` is not general enough
...
512 |         register_node!(graphene_core::text::TextGenerator<_, _, _>, input: graphene_core::EditorApi, params: [String, graphene_core::text::Font, f64]),
    |         ---------------------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: `TextGenerator<ClonedNode<std::string::String>, ClonedNode<graphene_core::text::Font>, ClonedNode<f64>>` must implement `graphene_core::Node<'0, graphene_core::EditorApi<'_>>`, for any lifetime `'0`...
    = note: ...but it actually implements `graphene_core::Node<'1, graphene_core::EditorApi<'1>>`, for some specific lifetime `'1`
    = note: this error originates in the macro `register_node` (in Nightly builds, run with -Z macro-backtrace for more info)
```
Which was somewhat confusing as I would have assumed that it could be the 'input lifetime.

Reply:
That's unfortunate, ran into similar issues with the `CloneNode`. Did you attempt to put the lifetime into the fn generics?
```rust
#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
pub struct CloneNode<O> {
	_o: PhantomData<O>,
}
#[node_macro::node_fn(CloneNode<_O>)]
fn clone<'n: 'input, _O: Clone>(input: &'n _O) -> _O {
	input.clone()
}

```